crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -234,8 +239,12 @@ impl<T: TransactionOrdering> TxPool<T> {
                     self.pending_pool.update_blob_fee(self.all_transactions.pending_fees.blob_fee);
                 for tx in removed {
                     let to = {
-                        let tx =
-                            self.all_transactions.txs.get_mut(tx.id()).expect("tx exists in set");
+                        let tx = self
+                            .all_transactions
+                            .txs
+                            .get_mut(&tx.id().sender)
+                            .and_then(|sender_txs| sender_txs.get_mut(tx.id()))
+                            .expect("tx exists in set");
```

Comment:
this we also have a few times now, could we make this a new helper of alltransactions?